lib/trino-orc/src/main/java/io/trino/orc/reader/ByteColumnReader.java

```diff
@@ -77,7 +79,7 @@ public ByteColumnReader(Type type, OrcColumn column, LocalMemoryContext memoryCo
             throws OrcCorruptionException
     {
         this.type = requireNonNull(type, "type is null");
-        verifyStreamType(column, type, t -> t == TINYINT || t == INTEGER);
+        verifyStreamType(column, type, t -> t == TINYINT || t == SMALLINT || t == INTEGER || t == BIGINT);
```

Comment:
i though the purpose of "Convert ORC block within ColumnAdaptation" was to have readers simplified, so surprised to see this change here.

I think  I like the non-composed (without column adaptations) approach better though, so no changes requested here.